src/meetingreadinesschecker/DefaultMeetingReadinessChecker.ts

```diff
@@ -94,6 +94,9 @@ export default class DefaultMeetingReadinessChecker implements MeetingReadinessC
     const rampSec = 0.1;
     const maxGainValue = 0.1;
 
+    if (this.oscillatorNode) {
```

Comment:
Stopping the previous tone is one approach, but a better way would be to stop the corresponding tone only.

```
private playTone(...) {
  // Create AudioContext nodes as local variables

  return () => {
     // Stop and disconnect nodes
  };
}
```

// In the caller
```
let stopTone;
try {
  stopTone = this.playTone();
} catch (error) {
} finally {
  stopTone();
}
```

Reply:
NVM.. since it takes a `HTMLAudioElement` parameter, my suggestion can be error prone too if they pass the same audio element.